src/main/java/frc/robot/subsystems/intake/IntakeIOTalonFX.java

```diff
@@ -0,0 +1,148 @@
+package frc.robot.subsystems.intake;
+
+import com.ctre.phoenix6.configs.CurrentLimitsConfigs;
+import com.ctre.phoenix6.configs.TalonFXConfiguration;
+import com.ctre.phoenix6.controls.DutyCycleOut;
+import com.ctre.phoenix6.controls.TorqueCurrentFOC;
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.ctre.phoenix6.signals.InvertedValue;
+
+import edu.wpi.first.wpilibj.DigitalInput;
+import frc.lib.team6328.util.TunableNumber;
+
+public class IntakeIOTalonFX implements IntakeIO {
+  private TalonFX rightRollerMotor;
+  private TalonFX leftRollerMotor;
+  private TalonFX drumMotor;
+  private final DigitalInput rightRollerIRSensor;
+  private final DigitalInput leftRollerIRSensor;
+  private final DigitalInput drumIRSensor;
+
+  // tunable numbers for roller and drum pid
+  private final TunableNumber rightRollerMotorKP =
+      new TunableNumber("Intake/rightRollerMotorKP", IntakeConstants.INTAKE_RIGHT_ROLLER_MOTOR_KP);
+  private final TunableNumber rightRollerMotorKI =
+      new TunableNumber("Intake/rightRollerMotorKI", IntakeConstants.INTAKE_RIGHT_ROLLER_MOTOR_KI);
+  private final TunableNumber rightRollerMotorKD =
+      new TunableNumber("Intake/rightRollerMotorKD", IntakeConstants.INTAKE_RIGHT_ROLLER_MOTOR_KD);
+
+  private final TunableNumber leftRollerMotorKP =
+      new TunableNumber("Intake/leftRollerMotorKP", IntakeConstants.INTAKE_LEFT_ROLLER_MOTOR_KP);
+  private final TunableNumber leftRollerMotorKI = 
+      new TunableNumber("Intake/leftRollerMotorKI", IntakeConstants.INTAKE_LEFT_ROLLER_MOTOR_KI);
+  private final TunableNumber leftRollerMotorKD = 
+      new TunableNumber("Intake/leftRollerMotorKD", IntakeConstants.INTAKE_LEFT_ROLLER_MOTOR_KD);
+
+  private final TunableNumber drumMotorKP =
+      new TunableNumber("Intake/drumMotorKP", IntakeConstants.INTAKE_DRUM_MOTOR_KP);
+  private final TunableNumber drumMotorKI =
+      new TunableNumber("Intake/drumMotorKI", IntakeConstants.INTAKE_DRUM_MOTOR_KI);
+  private final TunableNumber drumMotorKD = 
+      new TunableNumber("Intake/drumMotorKD", IntakeConstants.INTAKE_DRUM_MOTOR_KD);
+
+
+  public IntakeIOTalonFX() {
+    rightRollerIRSensor = new DigitalInput(IntakeConstants.INTAKE_RIGHT_ROLLER_IR_SENSOR_ID);
+    leftRollerIRSensor = new DigitalInput(IntakeConstants.INTAKE_LEFT_ROLLER_IR_SENSOR_ID);
+    drumIRSensor = new DigitalInput(IntakeConstants.INTAKE_DRUM_IR_SENSOR_ID);
+
+    configureIntakeMotors();
+  }
+
+  @Override
+  public void updateInputs(IntakeIOInputs inputs) {
```

Comment:
Invoke the refreshAll command on all status signals at once which is more efficient than on each signal:

```
BaseStatusSignal.refreshAll(
          leftRollerVelocityStatusSignal,
          leftRollerVelocityReferenceStatusSignal,
          ...);
```